geode-core/src/main/java/org/apache/geode/distributed/internal/direct/DirectChannel.java

```diff
@@ -142,7 +143,23 @@ public DirectChannel(Membership<InternalDistributedMember> mgr,
       props.setProperty("membership_port_range_start", "" + range[0]);
       props.setProperty("membership_port_range_end", "" + range[1]);
 
-      this.conduit = new TCPConduit(mgr, port, address, isBindAddress, this, bufferPool, props);
+      InetAddress conduitAddress = address;
+      if (!dc.getMembershipBindAddress().isEmpty()) {
+        try {
+          if (dc.getMembershipBindAddress().equals("*")) {
```

Comment:
Should "*" be the DEFAULT_MEMBERSHIP_BIND_ADDRESS?
Whether or not that's true, I think this special value for the setting should be described in ConfigurationProperties.

Reply:
If "*" were the default address for the bindings in DirectChannel and GMSHealthMonitor, it will simplify the whole solution. I created a PR to test if binding to all addresses by default would break any public test and just one distributedTest failed, although it works locally for me.

I will add a commit removing the no needed changes (which will be almost everything) so you can give me your opinion.

Reply:
@bschuchardt There you are: https://github.com/apache/geode/pull/6066

Reply:
I received some feedback from @Bill and he thinks that changing the default value introduces a security hole, so I have closed the PR where I was testing that.